src/components/dashboard/FaceVerification/screens/VerificationScreen.js

```diff
@@ -24,6 +26,8 @@ import {
   FV_ZOOMFAILED,
 } from '../../../../lib/analytics/analytics'
 
+const AB = random(0.5)
```

Comment:
As i remember from your discussion with @sirpy, AB state should be app-wide.
In this cases, and, accordingly to the React rules, the better option would be to use Context configurable via env vars:

```javascript
/** in config.js */
abRate: Number(process.env.REACT_APP_AB_RATE || 0.5)

/** abTesting.js */
import { createContext, useContext } from 'react'
import Config from 'path/to/config'

// you actually do not need export context, create provider and include it somewhere in the App component
// because the defaultValue argument will be used when a component does not have a matching Provider above it in the tree. 
// which will be enough for this particular task
const ABContext = createContext(Math.random() < Config.abRate ? 'A' : 'B')
const useABTesting = () => useContext(ABContext)

// se we'll export the hook only
export default useABTesting

/** some component */
import useABTesting from 'path/to/abTesting'

const AB = useABTesting()

Reply:
also useABTesting() hook could be included directly to the component selecting a or b view. for example, directly to the Instructions (you now select a or b in the FaceVerificationScreen and pass it down via props, using context would help you avoid that)

Reply:
also you could add useABComponent hook selecting one of two component (as in my suggestions before)

```javascript
const useABComponent = (componentA, componentB) => useContext(ABContext) === 'A' ? componentA : componentB
```

Reply:
according to @sirpy we need to choose between version A/B on each screen separately

Reply:
@omerzam okay then you could create the following hook

```javascript
const useABTesting = (componentA, componentB, abRate = Config.abRate) => Math.random() < abRate  ? ['A', componentA] : ['B', componentB]


// usage
const [ab, InstructionsComponent] = useABTesting(InstructionsA, InstructionsB)
```

Reply:
@serdiukov-o-nordwhale is there anything wrong with the way I implemented it?
what is the added value of this implementation?

Reply:
its not required. plus it is not a fixed random result for the whole user session.

Reply:
@sirpy but the current implementation also doesn't fixes result for the whole session.
if you need to fix result for the session you need to use context. or at least global var like FaceTecGlobalState